### PR TITLE
Adds a toggle on Settings page to show/hide the widget opener.

### PR DESCRIPTION
**Pull request by Shrikant9**
Issue: [#494](https://github.com/bigbinary/acehelp/issues/494)

### CODE REVIEW

**Comment by Shrikant9** [comment]
@breezyboa _a

**Comment by breezyboa** [comment]
@Shrikant9 _a we need to save the settings in the backend and not just modify the code snippet.
If we only modify the code snippet then the user would have to update the code snippet in the website also for the changes to take effect. The scope of this issue is to update the UI and save the settings in the backend. So there is rails/graphql work also involved here.

There is a separate issue #495 where it is expected that the widget would fetch the settings and then render accordingly

cc: @tejaswinichile

**Comment by Shrikant9** [comment]
@tejaswinichile _a
Please add the required GraphQL query so that I can integrate it in the Elm code.

**Comment by tejaswinichile** [comment]
@Shrikant9 _a Please check, added mutation to save the setting for the organization.

**Comment by neerajdotname** [comment]
@Shrikant9 Can you take care of the comment.

**Comment by Shrikant9** [comment]
@tejaswinichile I don't think just `changeVisibilityOfWidget` mutation will be enough. 

I also need a way to retrieve previously saved setting when I land on this page. Can you also add that query? I'll integrate it on the elm side.

**Comment by tejaswinichile** [comment]
@ritesh404 _a Please check .

File: app/javascript/packs/client.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var AceHelp = (function() {
     var internal = {};
     var _app;
 
-    function _insertWidget({ apiKey }) {
+    function _insertWidget({ apiKey }, showWidget) {
```
Comment (ritesh404) [inline app/javascript/packs/client.js]: We dont need this. Widget should fetch settings and render accordingly

File: app/javascript/packs/script-template.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ script.async = true;
 script.onload = function() {
     var _ace = window.AceHelp;
     if (_ace) {
-        _ace._internal.insertWidget({ apiKey: apiKey });
+        _ace._internal.insertWidget({ apiKey: apiKey }, true);
```
Comment (ritesh404) [inline app/javascript/packs/script-template.js]: This needs to be reverted back

File: app/assets/stylesheets/admin/app.scss [hunk 3]
```diff
@@ -123,4 +120,7 @@ $blueThemeColor: rgb(60, 170, 249);
     z-index: 1;
   }
 
+  .content-section {
+    margin: 20px 0px;
```
Comment (ritesh404) [inline app/assets/stylesheets/admin/app.scss]: We don't need this

File: test/graphql/mutations/widget_setting_mutation_test.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    assert_equal result.data.save_organization_widget_setting.setting.base_url, "http://bigbinary.com"
+  end
+
+  test "update widget setting visibility" do
+    query = <<-'GRAPHQL'
+              mutation($input: EnableDisableWidgetInput!) {
+                changeVisibilityOfWidget(input: $input) {
+                  setting {
+                    id
+                    base_url
```
Comment (ritesh404) [inline test/graphql/mutations/widget_setting_mutation_test.rb]: @tejaswinichile what does the base_url here refer to?

**Comment by Shrikant9** [comment]
@ritesh404 _a
I have added the missing text contents. 
The design doesn't exactly match the given wireframe but it's as close as I could get. For example, the green color seems a bit off, so I used the blue color to match our theme.

File: app/javascript/Admin/Page/Settings.elm [hunk 5]
```diff
@@ -35,52 +45,161 @@ init =
...
+            settingStatus visibility
+    in
     "<script>"
         ++ "var req=new XMLHttpRequest,baseUrl='"
         ++ baseUrl nodeEnv appUrl
         ++ "',apiKey='"
         ++ organizationKey
-        ++ "',script=document.createElement('script');script.type='text/javascript',script.async=!0,script.onload=function(){var e=window.AceHelp;e&&e._internal.insertWidget({apiKey:apiKey})};var link=document.createElement('link');link.rel='stylesheet',link.type='text/css',link.media='all',req.responseType='json',req.open('GET',baseUrl+'/packs/manifest.json',!0),req.onload=function(){var e=document.getElementsByTagName('script')[0],t=req.response;link.href=baseUrl+t['client.css'],script.src=baseUrl+t['client.js'],e.parentNode.insertBefore(link,e),e.parentNode.insertBefore(script,e)},req.send();"
+        ++ "',script=document.createElement('script');script.type='text/javascript',script.async=!0,script.onload=function(){var e=window.AceHelp;e&&e._internal.insertWidget({apiKey:apiKey},'"
+        ++ widgetToggleVisibility
```
Comment (ritesh404) [inline app/javascript/Admin/Page/Settings.elm]: @Shrikant9  This needs to be reverted as well

File: test/graphql/mutations/widget_setting_mutation_test.rb (hunk 4 above)
Comment (tejaswinichile) [reply test/graphql/mutations/widget_setting_mutation_test.rb]: We are just fetching it, I saw variables on setting page `base_url`, `api_key` and `visibility`. I have created these columns. Discussed this with @Shrikant9 before creating it.

**Pull request merged**